src/ryu.jl

```diff
@@ -6,8 +6,11 @@ function plain_precision_heuristic(xs::AbstractArray{<:AbstractFloat})
     ys = filter(isfinite, xs)
     precision = 0
     for y in ys
-        b, e10 = Ryu.reduce_shortest(convert(Float32, y))
-        precision = max(precision, -e10)
+        if isapprox(y, 0, atol=1e-16) # 0 has undefined rtol
```

Comment:
@timholy What about now?
The probelm with 0.0 is that it has undefined rtol and it is hard to judge if a number is close to zero
The choice behind 1e-16 is somewhat arbitrary

Reply:
My suggestion was to keep track of both the min and max of `e10`. Maybe something like this:
```julia
e10max = -(e10min = typemax(Int))
for y in ys
    _, e10 = Ryu.reduce_shortest(y)
    e10min = min(e10min, e10)
    e10max = max(e10max, e10)
end
precision = min(-e10min, -e10max+16)
```
That way if *all* the numbers are tiny, you won't truncate unnecessarily.

Not sure whether the old 0 was important for something, but this design would also allow you to incorporate that if necessary.